staff/edit-courses/edit-course.js

```diff
@@ -0,0 +1,359 @@
+fetch("get-courses.php").then((source) => source.json()).then(function(data) {
+    courses = data;
+    course_codes = data.map((value) => value.course_code);
+});
+
+// Add autocomplete functionality to a requisite input box
+function updateReqAutoComplete(req_num) {
+    let req_id = "reqCode-" + req_num;
+    let autoCompleteJS = new autoComplete({
+        selector: "#" + req_id, 
+        placeHolder: "EECS 101",
+        data: {
+                src: courses,
+                keys: ["course_code", "title"]
+            },
+        resultsList: {
+            element: (list, data) => {
+                const info = document.createElement("p");
+                if (data.results.length > 0) {
+                    info.innerHTML = `Displaying <strong>${data.results.length}</strong> out of <strong>${data.matches.length}</strong> results`;
+                } else {
+                    info.innerHTML = `Found <strong>${data.matches.length}</strong> matching results for <strong>"${data.query}"</strong>`;
+                }
+                list.prepend(info);
+            },
+            noResults: true,
+            maxResults: 15,
+            tabSelect: true
+        },
+        resultItem: {
+            element: (item, data) => {
+                item.style = "display: flex; justify-content: space-between;";
+                item.innerHTML = `
+                <span style="white-space: nowrap; overflow: hidden;">
+                ${data.value.course_code}
+                </span>
+                <span style="white-space: nowrap; width:50%; overflow: hidden; text-overflow: ellipsis; align-items: left; font-size: 13px; font-weight: 100; text-transform: uppercase;">
+                ${data.value.title}
+                </span>`;
+            }
+        },
+        events: {
+            input: {
+                focus: () => {
+                    if (autoCompleteJS.input.value.length) autoCompleteJS.start();
+                }
+            }
+        }
+    });
+    autoCompleteJS.input.addEventListener("selection", function (event) {
+        const feedback = event.detail;
+        autoCompleteJS.input.blur();
+        const selection = feedback.selection.value['course_code'];
+        document.querySelector("#" + req_id).innerHTML = selection;
+        autoCompleteJS.input.value = selection;
+    });
+    document.getElementById(req_id).addEventListener('change', function() {
+        document.getElementById(req_id).style.border = "";
+    });
+}
+
+function expandText(e, full_text, click, delta=DELTA, expand_=null) {
+    if (expand_timer) {  
+        clearTimeout(expand_timer);
+    }
+    if (e.target.innerText.endsWith(SUFFIX))
+    {
+        e.target.innerText = e.target.innerText.substring(0, e.target.innerText.length-3);
+    }
+
+    let current_len = e.target.innerText.length;
+    let target_len = full_text.trim().length;
+
+    if (click) {
+        expand = current_len != target_len;
+    }
+    if (expand_ !== null) {
```

Comment:
Perhaps instead of having a dummy variable, line 84 could be `if (click & expand === null)` and the parameter could also named `expand` then